packages/assets-controllers/src/DeFiPositionsController/DeFiPositionsController.ts

```diff
@@ -0,0 +1,248 @@
+  typeof controllerName,
+  DeFiPositionsControllerState
+>;
+
+export type DeFiPositionsControllerEvents =
+  DeFiPositionsControllerStateChangeEvent;
+
+export type DeFiPositionsControllerStateChangeEvent =
+  ControllerStateChangeEvent<
+    typeof controllerName,
+    DeFiPositionsControllerState
+  >;
+
+/**
+ * The external actions available to the {@link DeFiPositionsController}.
+ */
+export type DeFiPositionsControllerAllowedActions =
+  AccountsControllerListAccountsAction;
+
+/**
+ * The external events available to the {@link DeFiPositionsController}.
+ */
+export type DeFiPositionsControllerAllowedEvents =
+  | KeyringControllerUnlockEvent
+  | KeyringControllerLockEvent
+  | TransactionControllerTransactionConfirmedEvent
+  | AccountsControllerAccountAddedEvent;
+
+/**
+ * The messenger of the {@link DeFiPositionsController}.
+ */
+export type DeFiPositionsControllerMessenger = RestrictedMessenger<
+  typeof controllerName,
+  DeFiPositionsControllerActions | DeFiPositionsControllerAllowedActions,
+  DeFiPositionsControllerEvents | DeFiPositionsControllerAllowedEvents,
+  DeFiPositionsControllerAllowedActions['type'],
+  DeFiPositionsControllerAllowedEvents['type']
+>;
+
+/**
+ * Controller that stores assets and exposes convenience methods
+ */
+export class DeFiPositionsController extends StaticIntervalPollingController()<
+  typeof controllerName,
+  DeFiPositionsControllerState,
+  DeFiPositionsControllerMessenger
+> {
+  readonly #fetchPositions: (
+    accountAddress: string,
+  ) => Promise<DefiPositionResponse[]>;
+
+  readonly #isEnabled: () => boolean;
+
+  /**
+   * Tokens controller options
+   *
+   * @param options - Constructor options.
+   * @param options.messenger - The controller messenger.
+   * @param options.isEnabled - Function that returns whether the controller is enabled. (default: () => true)
+   */
+  constructor({
+    messenger,
+    isEnabled = () => true,
+  }: {
+    messenger: DeFiPositionsControllerMessenger;
+    isEnabled?: () => boolean;
+  }) {
+    super({
+      name: controllerName,
+      metadata: controllerMetadata,
+      messenger,
+      state: getDefaultDefiPositionsControllerState(),
+    });
+
+    this.setIntervalLength(TEN_MINUTES_IN_MS);
+
+    this.#fetchPositions = buildPositionFetcher();
+    this.#isEnabled = isEnabled;
+
+    this.messagingSystem.subscribe('KeyringController:unlock', () => {
+      this.startPolling(null);
+    });
+
+    this.messagingSystem.subscribe('KeyringController:lock', () => {
+      this.stopAllPolling();
+    });
+
+    this.messagingSystem.subscribe(
+      'TransactionController:transactionConfirmed',
+      async (transactionMeta) => {
+        if (!this.#isEnabled()) {
+          return;
+        }
+
+        await this.#updateAccountPositions(transactionMeta.txParams.from);
+      },
+    );
+
+    this.messagingSystem.subscribe(
+      'AccountsController:accountAdded',
```

Comment:
Should this controller also subscribe to the `:accountRemoved` event?

Reply:
Not needed. We are not persisting that state, so the next time we fetch new positions any removed accounts will no longer have data.